maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gittest/src/main/java/org/apache/maven/scm/provider/git/command/checkout/GitSshCheckOutCommandTckTest.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.sshd.git.GitLocationResolver;
+import org.apache.sshd.git.pack.GitPackCommandFactory;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.pubkey.KeySetPublickeyAuthenticator;
+import org.apache.sshd.server.auth.pubkey.PublickeyAuthenticator;
+import org.apache.sshd.server.session.ServerSession;
+import org.apache.sshd.util.test.CommonTestSupportUtils;
+import org.apache.sshd.util.test.CoreTestSupportUtils;
+import org.bouncycastle.openssl.PKCS8Generator;
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+import org.bouncycastle.openssl.jcajce.JcaPKCS8Generator;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.junit.Assume;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ *
+ */
+public abstract class GitSshCheckOutCommandTckTest
+    extends CheckOutCommandTckTest
+{
+    protected final SshServer sshServer;
+    protected final KeyPair keyPair;
+    protected final List<PublicKey> acceptedPublicKeys;
+
+    @Rule
+    public TemporaryFolder tmpDirectory = new TemporaryFolder();
+
+    protected GitSshCheckOutCommandTckTest() throws GeneralSecurityException
+    {
+         sshServer = CoreTestSupportUtils.setupTestServer( getClass() );
+         keyPair = CommonTestSupportUtils.generateKeyPair( KeyUtils.RSA_ALGORITHM, 2048 );
+         acceptedPublicKeys = new ArrayList<>();
+         acceptedPublicKeys.add( keyPair.getPublic() );
+         PublickeyAuthenticator authenticator = new KeySetPublickeyAuthenticator( "onlykey",
+                  acceptedPublicKeys );
+         sshServer.setPublickeyAuthenticator( authenticator );
+    }
+
+    void writePrivateKeyAsPkcs8( Path file, String passphrase )
+            throws IOException, GeneralSecurityException
+    {
+        // encryption only optional
+        if ( passphrase != null )
+        {
+            // encryption with format outlined in https://dnaeon.github.io/openssh-private-key-binary-format/
+            OpenSSHKeyPairResourceWriter writer = new OpenSSHKeyPairResourceWriter();
+            OpenSSHKeyEncryptionContext context = new OpenSSHKeyEncryptionContext();
+            context.setCipherType( "192" );
+            context.setPassword( passphrase );
+            try ( OutputStream output = Files.newOutputStream( file ) )
+            {
+                writer.writePrivateKey( keyPair, "comment", context, output );
+            }
+        }
+        else 
+        {
+            // wrap unencrypted private key as regular PKCS8 private key
+            PKCS8Generator pkcs8Generator = new JcaPKCS8Generator( keyPair.getPrivate(), null );
+            PemObject pemObject = pkcs8Generator.generate();
+            
+            try ( Writer writer = Files.newBufferedWriter( file );
+                            JcaPEMWriter pw = new JcaPEMWriter( writer ) )
+            {
+                pw.writeObject( pemObject );
+            }
```

Comment:
I am confused and what if you use the abover without a passowrd? I mean it is a setter on the context, therefore, optional...

Reply:
I am not getting the question. I could figure out a format with an encrypted private key leveraging the `JcaPKCS8Generator` which was accepted by SSH or JGit SSH. Feel free to try it out yourself.

Reply:
What if you use the OpenSSHWriter even with a `null` passphrase. Will that work?